exchange/providers/providers.go

```diff
@@ -0,0 +1,87 @@
+package providers
```

Comment:
I think this should probably not live in the exchange subdirectory, maybe make a new top level directory (ideally, we can move it out of the go-ipfs repo, but that can wait)